charts/opentelemetry-collector/values.schema.json

```diff
@@ -131,7 +131,11 @@
         "create": {
           "description": "Specifies whether a configMap should be created (true by default).",
           "type": "boolean"
-        }
+        },
+        "existingName": {
+          "description": "Specifies an existing configMap to be mounted to the pod",
```

Comment:
is there ever a case when you want existingName set to a real value _and_ create true?

Reply:
I don't think so. When _create_ is true the name of the ConfigMap is generated via a helper function, meaning it won't come from values.

Reply:
Thats handled here https://github.com/open-telemetry/opentelemetry-helm-charts/pull/1105/files#diff-12c2446fb11e3b1d1120f6b547457ee4dc1420594649d1d6101060548e8a5afbR219

ExistingName field is only used when its not empty @jaronoff97

Reply:
yes, i know. I meant it would also be used [here](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-collector/templates/configmap.yaml#L1) to prevent that potential footgun

Reply:
Oh something like this

```
{{- if and (eq .Values.mode "daemonset") (.Values.configMap.create) (not .Values.configMap.existingName) -}}
```

Reply:
yep! that way you don't accidentally try to create a configmap that already exists :) NOTE: there are three files where this could happen right now.

Reply:
Updated https://github.com/open-telemetry/opentelemetry-helm-charts/pull/1105/commits/c75f3288566355bf6e6d072105462c6d23b4c5df

Reply:
This was a good discussion. Instead of doing the check in the templates, we should add a `fail` condition to `NOTES.txt` that fails the install if `create: true` and `existingName` is truthy.

Reply:
Might not need to do that part, this check mainly prevents generating unnecessary resources. Now the check is on the ConfigMap I don't think this case is possible from user perspective right?

Reply:
Adding the check to `NOTES.txt` prevents users from writing a misleading values.yaml.  It would outlaw

```
configMap:
  create: true
  existingName: "my-custom-config"
```

That values.yaml shouldn't be supported because it is unclear what happens.  The changes you made to the template results in `existingName` being ignored, which saves the user, but I think explicitly failing would be better.  We don't like allowing bad config to be passed to the chart.

Reply:
Makes sense what do you think about this message?

```
{{- if and .Values.configMap.create .Values.configMap.existingName }}
{{ fail "[ERROR] The opentelemetry-collector chart support one of configMap.create or configMap.existingName but not both" }}
{{ end }}
```

Example from helm template

```console
helm template charts/opentelemetry-collector \
  --set configMap.create=true \
  --set configMap.existingName="ok" \
  --set image.repository=gcr.io \
  --set mode=daemonset                                    
Error: execution error at (opentelemetry-collector/templates/NOTES.txt:32:3): [ERROR] The opentelemetry-collector chart support one of configMap.create or configMap.existingName but not both

Use --debug flag to render out invalid YAML
```

Reply:
Updated 1bca1b4eb5b69c6feaceb8eb7043d91679ac7244 @TylerHelmuth